K2Bridge/Visitors/Aggregations/AggregationDictionaryVisitor.cs

```diff
@@ -70,6 +71,47 @@ public string BuildBucketAggregationQuery(BucketAggregation bucketAggregation, B
             query.Append($"{KustoQLOperators.CommandSeparator} {KustoQLOperators.Summarize} {bucketAggregation.SubAggregationsKustoQL}");
             query.Append($"{definition.BucketExpression};");
 
+            // Create a table named "metadata", with one column per aggregation and dynamic value.
+            if (definition.Metadata != null)
+            {
+                query.Append(BuildMetadataQuery(definition.Metadata));
+            }
+
+            return query.ToString();
+        }
+
+        /// <summary>
+        /// Given a metadata dictionary, generates a Kusto QL statement creating the metadata table:
+        /// datatable(['2']:string) [dynamic(['val1','val2', 'val3'])] | as metadata;
+        /// </summary>
+        /// <param name="metadata">A metadata dictionary.</param>
+        /// <returns>A string containing the Kusto QL statement.</returns>
+        public string BuildMetadataQuery(Dictionary<string, string> metadata)
```

Comment:
For now metadata only has one column.
Do you expect to sometime need multiple columns?

Reply:
Yes, if we plan to use the same logic for Metrics. It means we will have one additional column per metric with metadata.

Reply:
Which metrics are expected to have metadata? will they still be on the same aggregation key?

Reply:
Metrics that need to use the metadata table would insert their own aggregation key. i.e. if the bucket aggregation key is '2', the percentiles metrics key could be '3'. Currently the metrics that use the column name trick are Percentiles and Extended Stats if I'm not mistaken.

Reply:
But for now we don't plan to change them to this, right? ( I guess we don't have much time?)